src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignerInfoTests.netcoreapp.cs

```diff
@@ -221,6 +266,103 @@ public static void SignerInfo_AddRemoveUnsignedAttributes_JoinCounterSignaturesA
             cms.CheckSignature(true);
         }
 
+        [Fact]
+        public static void SignerInfo_AddOneUnsignedAttributeToCounterSigner_Adds()
+        {
+            SignedCms cms = new SignedCms();
+            cms.Decode(SignedDocuments.OneRsaSignerTwoRsaCounterSigners);
+
+            Assert.Equal(2, cms.SignerInfos[0].UnsignedAttributes.Count);
+            Assert.Equal(2, cms.SignerInfos[0].CounterSignerInfos.Count);
+
+            SignerInfo counterSigner0 = cms.SignerInfos[0].CounterSignerInfos[0];
+            SignerInfo counterSigner1 = cms.SignerInfos[0].CounterSignerInfos[1];
+
+            Assert.Equal(0, counterSigner0.UnsignedAttributes.Count);
+            Assert.Equal(0, counterSigner1.UnsignedAttributes.Count);
+
+            AsnEncodedData attribute = CreateTimestampToken(1);
+            counterSigner0.AddUnsignedAttribute(attribute);
+
+            // we didn't modify existing instance
+            Assert.Equal(0, counterSigner0.UnsignedAttributes.Count);
+            Assert.Equal(0, counterSigner1.UnsignedAttributes.Count);
+
+            // verify we didn't modify anything we shouldn't
+            Assert.Equal(1, cms.SignerInfos.Count);
+            Assert.Equal(2, cms.SignerInfos[0].UnsignedAttributes.Count);
+            Assert.Equal(2, cms.SignerInfos[0].CounterSignerInfos.Count);
```

Comment:
There's probably value in doing this, in addition to having saved `cms.SIgnerInfos[0]` as a local and checking that that instance also doesn't see the change.